R/format_cfr_neatly.R

```diff
@@ -0,0 +1,53 @@
+#' Neatly format CFR estimates
+#'
+#' @description Format the CFR MLE and 95% range estimates neatly, giving the
+#' estimates to 3 significant figures and presenting the MLE and 95% interval
+#' on a single line
+#'
+#' @param cfr_in A named vector of the format outputted by [ccfr_uncertainty()]
+#' or [static_cfr()]
+#'
+#' @return Prints the output of [ccfr_uncertainty()] or [static_cfr()] neatly to
+#' the R terminal
+#' @export
+#'
+#' @examples
+#' # Get onset to death distribution from epiparameter
+#' onset_to_death_ebola <- epiparameter::epidist("ebola", "onset_to_death")$pmf
+#'
+#' # load Ebola 1976 outbreak data
+#' data("ebola1976")
+#' # Calculate static naive CFR
+#' ncfr <- static_cfr(df_in = ebola1976, correct_for_delays = FALSE)
+#'
+#' # Calculate static corrected CFRs
+#' ccfr <- static_cfr(
+#'   df_in = ebola1976,
+#'   correct_for_delays = TRUE,
+#'   delay_pmf = onset_to_death_ebola
+#' )
+#'
+#' # Formats the output of the CFR data.frames nicely
+#' # and prints to the terminal
+#' format_cfr_neatly(ncfr)
+#' format_cfr_neatly(ccfr)
+format_cfr_neatly <- function(cfr_in) {
+  if (is.data.frame(cfr_in)) {
+    stop("`cfr_in` only works with static estimates - you may have provided a
+    rolling CFR estimate")
+  }
+  stopifnot(
+    "`cfr_in` must be numeric vector with the names `cfr_me`, \
+    `cfr_low`, `cfr_high`" =
+      (is.numeric(cfr_in) &
+        all(utils::hasName(cfr_in, c("cfr_me", "cfr_low", "cfr_high")))
+      )
+  )
+
+  # combining the MLE estimate and the 95% interval neatly and
+  # giving each to only 3 significant figures
+  sprintf(
+    "CFR: %.3f%% (95%% Ci: %.3f%% -- %.3f%%)",
```

Comment:
Is the `--` text necessary vs `-`, which would parse better in documents?

Would suggest 2sf as default, perhaps with `signif()` function in future versions.
```suggestion
    "CFR: %.2f%% (95%% CI: %.2f%% - %.2f%%)",
```

Reply:
Not necessarily, but thought it might avoid confusion with a negative sign.

I've used the `%.3f` and removed `signif()` - is there a good reason to have it here?

Reply:
No, I just didn't know about the %.Xf notation! I'll ask on here, but I'd be interested to read some documentation about it, as I've not seen it before